test/helpers/kubectl.go

```diff
@@ -912,37 +912,91 @@ func (kub *Kubectl) CiliumReport(namespace string, commands ...string) {
+	netpols := kub.Exec(fmt.Sprintf(
+		"%s get netpol -o jsonpath='%s' --all-namespaces",
+		KubectlCmd, policiesFilter))
+	fmt.Fprintf(CheckLogs, "Netpols loaded: %v\n", netpols.Output())
+
+	cnp := kub.Exec(fmt.Sprintf(
+		"%s get cnp -o jsonpath='%s' --all-namespaces",
+		KubectlCmd, policiesFilter))
+	fmt.Fprintf(CheckLogs, "CiliumNetworkPolicies loaded: %v\n", cnp.Output())
+
+	cepFilter := `{range .items[*]}{.metadata.name}{"="}{.status.status.policy.realized.policy-enabled}{"\n"}{end}`
+	cepStatus := kub.Exec(fmt.Sprintf(
+		"%s get cep -o jsonpath='%s' --all-namespaces",
+		KubectlCmd, cepFilter))
+
+	fmt.Fprintf(CheckLogs, "Endpoint Policy Enforcement:\n")
+
+	table := tabwriter.NewWriter(CheckLogs, 5, 0, 3, ' ', 0)
+	for pod, policy := range cepStatus.KVOutput() {
+		fmt.Println(pod, "=>", policy)
+		fmt.Fprintf(table, "\t%s\t=>\t%s\n", pod, policy)
+	}
+	table.Flush()
+
+	var controllersFilter = `{range .controllers[*]}{.name}{"="}{.status.consecutive-failure-count}{"\n"}{end}`
+	for _, pod := range pods {
+		var prefix = ""
+		status := kub.CiliumExec(pod, "cilium status --all-controllers -o json")
+		result, _ := status.Filter(controllersFilter)
+		var total = 0
+		var failed = 0
+		for _, status := range result.KVOutput() {
+			total++
+			if status != "" {
+				failed++
+				prefix = "⚠️  "
+			}
+		}
+		statusFilter := `Status: {.cilium.state}  Health: {.cluster.ciliumHealth.state}` +
+			` Nodes "{.cluster.nodes[*].name}" ContinerRuntime: {.container-runtime.state}` +
+			` Kubernetes: {.kubernetes.state} KVstore: {.kvstore.state}`
+		data, _ := status.Filter(statusFilter)
+		fmt.Fprintf(CheckLogs, "%sCilium agent %q: %s Controllers: Total %d Failed %d\n",
+			prefix, pod, data, total, failed)
+	}
+}
+
 // ValidateNoErrorsOnLogs checks in cilium logs since the given duration (By
 // default `CurrentGinkgoTestDescription().Duration`) do not contain `panic`,
 // `deadlocks` or `segmentation faults` messages. In case of any of these
 // messages, it'll mark the test as failed.
 func (kub *Kubectl) ValidateNoErrorsOnLogs(duration time.Duration) {
+	var logs string
 	cmd := fmt.Sprintf("%s -n %s logs --timestamps=true -l k8s-app=cilium --since=%vs",
 		KubectlCmd, KubeSystemNamespace, duration.Seconds())
 	res := kub.Exec(fmt.Sprintf("%s --previous", cmd), ExecOptions{SkipLog: true})
-	if !res.WasSuccessful() {
-		res = kub.Exec(cmd, ExecOptions{SkipLog: true})
+	if res.WasSuccessful() {
+		logs += res.Output().String()
 	}
-	logs := res.Output().String()
-	for _, message := range checkLogsMessages {
-		gomega.ExpectWithOffset(1, logs).ToNot(gomega.ContainSubstring(message),
-			"Found a %q in Cilium logs", message)
+	res = kub.Exec(cmd, ExecOptions{SkipLog: true})
+	if res.WasSuccessful() {
+		logs += res.Output().String()
 	}
-}
 
-// CheckLogsForDeadlock checks if the logs for Cilium log messages that signify
-// that a deadlock has occurred.
-func (kub *Kubectl) CheckLogsForDeadlock() {
```

Comment:
Was this function never called? grep shows me some places it's used but I guess it's another version of it that isn't on the `Kubectl` type?

Reply:
No for Kubectl ;-)